src/main/java/ui/MoveMethodPanel.java

```diff
@@ -111,38 +115,58 @@ private JComponent createButtonsPanel() {
 
         doRefactorButton.setText(IntelliJDeodorantBundle.message(REFACTOR_BUTTON_TEXT_KEY));
         doRefactorButton.addActionListener(e -> refactorSelected());
-        model.addTableModelListener(l -> doRefactorButton.setEnabled(model.isAnySelected()));
         buttonsPanel.add(doRefactorButton);
 
         refreshButton.setText(IntelliJDeodorantBundle.message(REFRESH_BUTTON_TEXT_KEY));
         refreshButton.addActionListener(l -> refreshPanel());
+        refreshButton.setEnabled(true);
         buttonsPanel.add(refreshButton);
         panel.add(buttonsPanel, BorderLayout.EAST);
 
+        tableRefreshListener = l -> enableButtonsOnConditions();
+        model.addTableModelListener(tableRefreshListener);
+
         panel.add(info, BorderLayout.WEST);
         return panel;
     }
 
-    private void refactorSelected() {
+    private void enableButtonsOnConditions() {
+        doRefactorButton.setEnabled(model.isAnySelected());
+        selectAllButton.setEnabled(model.getRowCount() != 0);
+        deselectAllButton.setEnabled(model.isAnySelected());
+    }
+
+    private void disableButtonsButRefresh() {
```

Comment:
I suggest also disable `Refresh` button while calculating refactorings. It will avoid running multiple processes when `Refresh` button is selected several times in sequence.
Also we do not need a `Refresh` button while moving method.

Reply:
done

Reply:
Done.